[Feature] #7 UserEntity 작성

CodingScript990 opened the pull request:
1. UserEntity 작성
   - id, email, password, nickName, userName, phone, createdAt, updateAt, role, isActive, isDeleted, lastLoginAt(기본 회원가입 유형)
2. RoleEntity 작성
    - user, admin(회원인지, 슈퍼 관리자인지를 구분해주기 위한 작업)

CodingScript990 committed:
#4-security-environment - [MySQL 연결완료]

- Content : 
 1. mysql 연결 완료

CodingScript990 committed:
Merge branch '4-security-environment' into main

CodingScript990 committed:
Merge branch '2-security-environment' into main

CodingScript990 committed:
Revert "Merge branch '2-security-environment' into main"

This reverts commit 397748a2c7290626ba2572971eb12cb96a34e6ca.

CodingScript990 committed:
Merge branch '#2-security-environment' into main

CodingScript990 committed:
feat: 유저 Entity 작성

1. UserEntity 작성
   - id, email, password, nickName, userName, phone, createdAt, updateAt, role, isActive, isDeleted, lastLoginAt(기본 회원가입 유형)
2. RoleEntity 작성
    - user, admin(회원인지, 슈퍼 관리자인지를 구분해주기 위한 작업)

CodingScript990 committed:
feat: 유저 Entity 작성

1. UserEntity 작성
   - id, email, password, nickName, userName, phone, createdAt, updateAt, role, isActive, isDeleted, lastLoginAt(기본 회원가입 유형)
2. RoleEntity 작성
    - user, admin(회원인지, 슈퍼 관리자인지를 구분해주기 위한 작업)

CodingScript990 committed:
Merge branch 'feat/back/#7-user-entity' into develop

csumin0825 committed:
feat: 게시판 Entity 작성
  - Board Entity 작성(아티스트 id, 팀 id 주석처리)
  - BoardCategory Entity 작성

csumin0825 committed:
feat: 게시판 카테고리 테이블명 지정
  - board_category 테이블명 지정 어노테이션 추가

csumin0825 committed:
feat: 게시판 Entity 수정
  - created_at,updated_at 삭제

greedyBackEnd commented:
다대다 매핑 구조가 필요합니다!
createdAt 등은 BaseEntity에서 가져옵니다!

csumin0825 committed:
feat: 게시판 Entity 게시글 매핑
  - article 매핑 코드 추가

sy-k-98 committed:
feat : chat Entity 추가
- ChatRoom
- ChatMessage

CodingScript990 committed:
[Feature] #9 UserEntity 수정

- UserClass : 
      1. 유저 역할구분 타입을 Role 에서 String 타입으로 수정
      2. 각 Column명 수정
      3. Date type을 LocalDateTime으로 수정
      4. @CreatedDate 애노테이션을 이용하여 회원가입 날짜를 생성 되도록 수정, @LastModifiedDate 애노테이션을 이용하여 회원정보 수정날짜가 생성되도록 수정
      5. boolean 타입을 Boolean 타입으로 수정(isActive, isDeleted)
      6. lastLoginAt 컬럼 타입도 Date type 에서 LocalDateTime 타입으로 수정
      7. Role enum Class 삭제 => UserEntity Class에 Role type 을 String 타입으로 수정

CodingScript990 committed:
[Feature] #9 UserEntity 수정

- UserClass : 
      1. 유저 역할구분 타입을 Role 에서 String 타입으로 수정
      2. 각 Column명 수정
      3. Date type을 LocalDateTime으로 수정
      4. @CreatedDate 애노테이션을 이용하여 회원가입 날짜를 생성 되도록 수정, @LastModifiedDate 애노테이션을 이용하여 회원정보 수정날짜가 생성되도록 수정
      5. boolean 타입을 Boolean 타입으로 수정(isActive, isDeleted)
      6. lastLoginAt 컬럼 타입도 Date type 에서 LocalDateTime 타입으로 수정
      7. Role enum Class 삭제 => UserEntity Class에 Role type 을 String 타입으로 수정

CodingScript990 committed:
[Feature] #7 UserEntity 수정

- UserClass : 
      1. 유저 역할구분 타입을 Role 에서 String 타입으로 수정
      2. 각 Column명 수정
      3. Date type을 LocalDateTime으로 수정
      4. @CreatedDate 애노테이션을 이용하여 회원가입 날짜를 생성 되도록 수정, @LastModifiedDate 애노테이션을 이용하여 회원정보 수정날짜가 생성되도록 수정
      5. boolean 타입을 Boolean 타입으로 수정(isActive, isDeleted)
      6. lastLoginAt 컬럼 타입도 Date type 에서 LocalDateTime 타입으로 수정
      7. Role enum Class 삭제 => UserEntity Class에 Role type 을 String 타입으로 수정

CodingScript990 committed:
Merge remote-tracking branch 'origin/feat/back/#7-user-entity' into feat/back/#9-user-entity

CodingScript990 committed:
[Feature] #9 UserEntity 마무리

1. BaseEntity 를 상속받아서 createdAt, UpdateAt 사용할 수 있게 수정
2. PostMan 으로 Test한 결과 데이터가 잘 들어감을 확인함

CodingScript990 committed:
[Feature] #9 UserEntity 마무리

1. BaseEntity 를 상속받아서 createdAt, UpdateAt 사용할 수 있게 수정
2. PostMan 으로 Test한 결과 데이터가 잘 들어감을 확인함

greedyBackEnd committed:
feat : 팀, 아티스트 엔티티 작성
- artist(id, name, isSolo, description, profileUrl, nationality)
- team(id, name, description, logoUrl)

sy-k-98 committed:
feat : chat Entity TODO 주석 추가
- ChatRoom
- ChatMessage

sy-k-98 committed:
feat : chat Entity TODO 주석 추가
- ChatRoom
- ChatMessage

greedyBackEnd committed:
Merge pull request #15 from sy-k-98/feat/back/#3-create-chat-entity

[Feature] #3 ChatEntity 작성

greedyBackEnd committed:
Merge pull request #10 from csumin0825/feat/back/#5-board-entity

[Feature] #5 BoardEntity 작성

greedyBackEnd committed:
Merge pull request #14 from greedyBackEnd/feat/back/#12-team-artist-entitiy

[Feature] #12 팀, 아티스트 엔티티 작성

greedyBackEnd committed:
refactor : accessLevel 수정
- AccessLevel.PACKAGE -> AccessLevel.PROTECTED

greedyBackEnd committed:
refactor : createdBy, updatedBy 제거

CodingScript990 committed:
[Feature] #9 UserEntity 매핑작업[완료]

1.  다대다 매핑 구조 TODO 주석으로 각 필요한 매핑작업 완료
2. createdAt, updateAt 컬럼은 BaseEntity 에서 가져오도록 수정
3. 회원구분에서 Role Enum Class를 생성하여 User, Admin 인지 구분할 수 있게 수정

CodingScript990 committed:
[Feature] #9 UserEntity 매핑작업[완료]

1.  다대다 매핑 구조 TODO 주석으로 각 필요한 매핑작업 완료
2. createdAt, updateAt 컬럼은 BaseEntity 에서 가져오도록 수정
3. 회원구분에서 Role Enum Class를 생성하여 User, Admin 인지 구분할 수 있게 수정

greedyBackEnd committed:
refactor : createdBy, updatedBy 제거

CodingScript990 committed:
Merge remote-tracking branch 'Happi-Cat-Tiger/develop' into develop

greedyBackEnd committed:
Merge pull request #17 from greedyBackEnd/refactor/back/#16-base-entity

[Refactor] #16 Base Entity 수정

greedyBackEnd reviewed (commented):
- gitignore에 이미 yml에 대한 ignore이 추가되어있습니다.
- 장바구니 같은 경우 다대다 매핑이 아닌, 일대다/다대일 매핑으로 나누어야합니다. (+ 그 외 여러가지)
- boolean 칼럼에 대한 default 설정이 필요합니다.

CodingScript990 committed:
Merge remote-tracking branch 'Happi-Cat-Tiger/develop' into develop

CodingScript990 committed:
Merge remote-tracking branch 'Happi-Cat-Tiger/develop'

CodingScript990 committed:
Merge remote-tracking branch 'origin/feat/back/#7-user-entity' into feat/back/#9-user-entity

CodingScript990 committed:
[Feature] #9 UserEntity 요구사항 [수정]

1. gitignore 중복 작성한 yml file 제거
2. 장바구니 매핑 수정
3. Boolean Column default value 추가

CodingScript990 committed:
Merge branch 'feat/back/#9-user-entity' into develop

greedyBackEnd reviewed (commented):
pr하실 때 main이 아닌 develop으로 수정부탁드립니다.
createdAt 등 BaseEntity가 가지는 것은 빼버리셔도 됩니다.
table 네임이 user가 되면  좋겠습니다.
AccessLevel이 public이  아닌 PROTECTED가 되면 좋습니다.
boolean값에 대한  default 설정이 필요합니다. (@PrePersist 참고해보시면 좋을 것 같습니다.)
mappedBy에 대한 수정이 필요해 보입니다.

CodingScript990 committed:
[Feature] #9 UserEntity - 3차 요구사항[완료]

1. BaseEntity 상속받는 column들 제거완료
2. table 네이밍 user 로 변경완료
3. AccessLevel PROTECTED 로 변경완료
4. BOOLEAN 값에 대한 default 설정 완료
5. mappedBy 에대한 수정완료

CodingScript990 commented:
죄송해요.. 제가 develop branch로 변경 안되어 있는걸.. 이제 확인했습니다.. 지금은 develop branch로 변경했습니다. main으로 이번에 올라온점 사과드립니다. 요구하신 사항들 업로드 하였습니다. commit 전 한번 더 확인하겠습니다!
